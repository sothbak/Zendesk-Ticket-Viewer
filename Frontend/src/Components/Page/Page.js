import React, { Component } from 'react';
import List from '../List/List';
// import axios from 'axios';

class Page extends Component {

    state = {
        total_tickets : 0,
        ticket_data : null,
        loaded : false
    }

    async getTickets(page) {
        let response = await fetch('http://localhost:4001/tickets/');
        await response.json().then(data => {
            console.log(data);
            this.setState({total_tickets : data.count, ticket_data : data.tickets, loaded : true});
            // console.log(data.tickets[2].subject);
            // console.log(this.tickets[2]);
        })
    }
    componentDidMount() {
        this.getTickets();
    }

    render() {
        let list = null;
        this.state.loaded ? list = (<List ticket_data={this.state.ticket_data}/>) : list = null;
        return (
            <div>
                {/* <List tickets={this.state.ticket_data}/> */}
                {list}
            </div>
        )
    }
}

export default Page;