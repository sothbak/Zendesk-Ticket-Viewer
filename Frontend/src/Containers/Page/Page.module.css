.header {
    text-decoration: underline solid;
}

.list {
    position: absolute;
    float: left;
    width: 60%;
    background: rgb(117, 201, 166);
    height: 100%;
    overflow-y: scroll;
    /* border-right: 1px solid black; */
    /* padding-top: 10px;
    padding-bottom: 25px; */
}

.ticketDisplay {
    float: right;
    width: 40%;
    height: 100%;
}