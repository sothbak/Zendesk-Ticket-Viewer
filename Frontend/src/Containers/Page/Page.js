import React, { Component } from 'react';
import axios from 'axios';

import List from '../../Components/List/List';
import DisplayTicket from '../../Components/DisplayTicket/DisplayTicket';
import './Page.css';

class Page extends Component {

    state = {
        total_tickets : 0,
        page_tickets : 0,
        ticket_data : null,
        loaded : false,
        current_page : 1,
        total_pages : 1,
        showDisplayTicket : false,
        displayTicketData : null,
        connectedToServer : true,
    }

    getTickets = async (page) => {
        axios.get('http://localhost:4001/tickets/' + page)
        .then(response => {
            console.log(response.data);
            const data = response.data;
            this.setState({total_tickets : data.count, ticket_data : data.tickets, page_tickets : data.tickets.length, total_pages : Math.ceil(data.count/25), current_page : page, loaded : true});
        })
        .catch(error => {
            if (error.response) {
                // 4xx/5xx error
                console.log(error.response)
            }
            else {
                console.log("Couldn't connect to the server");
                console.log(error.response)
                this.setState({connectedToServer : false});
            }
        })
    }

    componentDidMount() {
        this.getTickets(this.state.current_page);
    }

    nextPage = () => {
        const current_page = this.state.current_page;
        this.setState({loaded : false}, () => {
            this.getTickets(current_page + 1);
        });
    }

    prevPage = () => {
        const current_page = this.state.current_page;
        this.setState({loaded : false}, () => {
            this.getTickets(current_page - 1);
        });
    }

    clickDisplayTicket = (id, subject, status, description) => {
        const displayTicket = {id : id, subject : subject, status : status, description : description};
        this.setState({showDisplayTicket : true, displayTicketData : displayTicket});
    }

    closeDisplayTicket = () => {
        this.setState({showDisplayTicket : false, displayTicketData : null});
    }

    render() {
        let errorMessage = null;
        let list = null;
        this.state.connectedToServer ? errorMessage = null : errorMessage = <h1>Sorry, couldn't connect to the server!</h1>
        
        // need key as when key updates, it updates the child component
        this.state.loaded ? list = (<List key={this.state.current_page} ticket_data={this.state.ticket_data} clickDisplayTicket={this.clickDisplayTicket} />) : list = null;
        
        let ticketShowing= null;
        this.state.showDisplayTicket ? ticketShowing = (<DisplayTicket data={this.state.displayTicketData} closeDisplayTicket={this.closeDisplayTicket} />) : ticketShowing = null;

        const nextButton = (<button disabled={!this.state.loaded || this.state.current_page === this.state.total_pages} onClick={this.nextPage}>Next page</button>);
        const prevButton = (<button disabled={!this.state.loaded || this.state.current_page === 1} onClick={this.prevPage}>Previous page</button>);

        return (
            <div>
                {ticketShowing}
                {prevButton}
                {nextButton}
                {errorMessage}               
                {list}
                <p>{this.state.total_tickets} total ticket(s), this page is showing {this.state.page_tickets} ticket(s)</p>
                <p>currently on page {this.state.current_page} of {this.state.total_pages}</p>
                {prevButton}
                {nextButton} 
            </div>
        )
    }
}

export default Page;