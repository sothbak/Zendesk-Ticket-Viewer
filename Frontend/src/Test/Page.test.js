import React from 'react';
import { shallow } from 'enzyme';
import Page from '../Containers/Page/Page.js';
describe('Page', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Page />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('should render correctly with list of ticket data', () => {
  //   const page = shallow(<Page />);
  //   page.setState({ticket_data : data});
  // });

});