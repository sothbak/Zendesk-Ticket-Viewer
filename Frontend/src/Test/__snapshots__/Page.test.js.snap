// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly 1`] = `
<div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Previous page
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Next page
  </button>
  <p>
    0
     total ticket(s), this page is showing 
    0
     ticket(s)
  </p>
  <p>
    currently on page 
    1
     of 
    1
  </p>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Previous page
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Next page
  </button>
</div>
`;
